### PR TITLE
/monitoring/tactical: fall back to unknown services if none critical

### LINKED ISSUE #4494: Tactical overview: include unknown services in counter
## Is your feature request related to a problem? Please describe.

In contrast to unreachable hosts unknown services aren’t just collateral damage which goes away once all down hosts are up again.

## Describe the solution you'd like

The right counter in /monitoring/tactical, its color and description includes unknown services.

## Describe alternatives you've considered

The mentioned counter falls back to unknown services if none are critical and unhandled.

### PR DESCRIPTION
**Pull request by Al2Klimov**
fixes #4494

### CODE REVIEW

**Comment by Al2Klimov** [comment]
## Before

Bildschirmfoto 2021-08-02 um 14 03 54
Bildschirmfoto 2021-08-02 um 14 04 41

## After

Bildschirmfoto 2021-08-02 um 14 13 10
Bildschirmfoto 2021-08-02 um 14 13 40
Bildschirmfoto 2021-08-02 um 14 14 14

File: modules/monitoring/application/controllers/TacticalController.php [hunk 1]
```diff
@@ -92,9 +92,11 @@ public function indexAction()
             ->addSlice($summary->services_unknown_unhandled, array('class' => 'slice-state-unknown'))
             ->addSlice($summary->services_pending, array('class' => 'slice-state-pending'))
             ->addSlice($summary->services_pending_not_checked, array('class' => 'slice-state-not-checked'))
-            ->setLabelBig($summary->services_critical_unhandled)
-            ->setLabelBigEyeCatching($summary->services_critical_unhandled > 0)
-            ->setLabelSmall($this->translate('Services Critical'));
+            ->setLabelBig($summary->services_critical_unhandled ?: $summary->services_unknown_unhandled)
+            ->setLabelBigEyeCatching($summary->services_critical_unhandled + $summary->services_unknown_unhandled > 0)
+            ->setLabelSmall($summary->services_critical_unhandled > 0 || $summary->services_unknown_unhandled < 1
```
Comment (nilmerg) [inline modules/monitoring/application/controllers/TacticalController.php:97]: Can't you just use `$summary->services_unknown_unhandled > 0 ? $this->translate('Services Unknown') : $this->translate('Services Critical')`?

**Review by nilmerg — CHANGES_REQUESTED** [review]
* As shown in your screenshot, red is not the color for UNKNOWN. It's purple.

File: modules/monitoring/application/controllers/TacticalController.php (hunk 1 above)
Comment (Al2Klimov) [reply modules/monitoring/application/controllers/TacticalController.php:97]: Imagine there are some critical and some unknown. And now have a look at what I pass to setLabelBig()... TL;DR: no, not really.

Comment (nilmerg) [reply modules/monitoring/application/controllers/TacticalController.php:97]: I see.

**Comment by nilmerg** [comment]
> Also I doubt the unknown color is eye-catching enough with the dark theme.

Doesn't matter. Red is *not* the color for UNKNOWN. Period. If that doesn't look okay with the dark theme, then open a new issue and let us handle that as part of v2.10 (theme sprint).

**Comment by nilmerg** [comment]
Please update the screenshots

**Comment by Al2Klimov** [comment]
Bildschirmfoto 2021-08-03 um 10 53 09
Bildschirmfoto 2021-08-03 um 10 53 47
Bildschirmfoto 2021-08-03 um 10 54 35

File: library/Icinga/Chart/Donut.php [hunk 2]
```diff
@@ -332,9 +356,9 @@ protected function assemble()
                         'attributes' => array(
                             'aria-label' => $labelBig . ' ' . $labelSmall,
                             'href' => $this->getLabelBigUrl() ? $this->getLabelBigUrl()->getAbsoluteUrl() : null,
-                            'class' => $this->labelBigEyeCatching
-                                ? 'donut-label-big donut-label-big-eye-catching'
-                                : 'donut-label-big'
+                            'class' => $this->labelBigState === null
+                                ? 'donut-label-big'
+                                : 'donut-label-big fg-stateful state-' . $this->labelBigState
```
Comment (nilmerg) [inline library/Icinga/Chart/Donut.php:361]: `fg-stateful` is a mixin, not a class. Use it inside the definition for `donut-label-big` instead. Also, please remove the `donut-label-big-eye-catching` rule as it's obsolete now.

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Pull request merged**